### ISSUE TITLE
Watchdog is not defined in prod build

### ISSUE THREAD

**Issue by martajedrecka** [ckeditor/ckeditor5-angular#194]
There is an error: Watchdog is not defined when the app is build (both build with --prod flag and without). The problem doesn't exists on local (`ng serve`) 

![image](https://user-images.githubusercontent.com/15836906/79915988-41758300-8428-11ea-9793-8d83fe60376b.png)

    "@ckeditor/ckeditor5-angular": "^1.2.3",
    "@ckeditor/ckeditor5-build-classic": "^18.0.0",
    "@ckeditor/ckeditor5-watchdog": "^18.0.0",
    "@angular/core": "8.2.2",

I've added package for watchdog, but in my code I'm not using Watchdog. Simple reconstruction of the example provided in docs is implemented: that is editor with onChange function.

**Comment by ma2ciek** [comment]
Hi @martajedrecka,

Thanks for the issue, I'll try to reproduce it soon.

**Comment by martajedrecka** [comment]
hello, when can we expect some feedback about that issue? We were looking for some production ready solution and we don't have much time for waiting. It's a pity that everything what was implemented locally and what works like a charm will have to go to trash

**Comment by ma2ciek** [comment]
Hi, sorry for the lack of response.

I've just run an app with `ng build --prod` which includes the `ckeditor5-angular` component and had no issues.

Maybe you have something specified differently? I'll post a configuration with which I'm testing this component.

> It's a pity that everything what was implemented locally and what works like a charm will have to go to trash

That's unfortunately pretty common for frameworks like Angular where the dev and prod environments diff much :/ 

That import where the app breaks is internal - the `EditorWatchdog` imports the `Watchdog` class, but it's hard to tell why it breaks in your app.

**Comment by pmacn** [comment]
Have the same issue here. The issue goes away if you install ng2-ckeditor. This I obviously not ideal or even a workable solution in most cases but it might give some insight into the problem

**Comment by ma2ciek** [comment]
Hi!

Is it possible for you to create a repository with steps to reproduce the error? I tried it once but I failed to reproduce it. Also, could you share your `tsconfig.json` file? I guess that this error might occur because of some kind of incorrect TypeScript transpilation.

**Comment by jbgtmartin** [comment]
Hi @pmacn, can you explain a bit more what you mean by "import the module from ng2-ckeditor"? Where exactly do you import the module? Thanks!

**Comment by jbgtmartin** [comment]
I have the exact same issue, except that I'm able to build without --prod (`ng build`). I only see the error whith `ng build --prod`.

**Comment by ma2ciek** [comment]
@jbgtmartin, could you share your TS / Angular configuration? You can pin an older version (https://github.com/ckeditor/ckeditor5-angular/releases/tag/v1.1.2) that doesn't import the `ckeditor5-watchdog` package. 

I guess that this has to be somehow connected with a different TS resolution system.

**Comment by jbgtmartin** [comment]
Hi, thanks for your answer! As expected, it's working fine with v1.1.2.

Here is my config:

#### package.json

```js
{
  ...
  "dependencies": {
    "@angular/animations": "^8.0.0",
    "@angular/cdk": "^8.2.3",
    "@angular/common": "~8.0.0",
    "@angular/compiler": "~8.0.0",
    "@angular/core": "~8.0.0",
    "@angular/flex-layout": "^8.0.0-beta.27",
    "@angular/forms": "~8.0.0",
    "@angular/material": "^8.2.3",
    "@angular/platform-browser": "~8.0.0",
    "@angular/platform-browser-dynamic": "~8.0.0",
    "@angular/router": "~8.0.0",
    "@ckeditor/ckeditor5-angular": "^1.2.3",
    "@ckeditor/ckeditor5-build-classic": "...", // Our custom build, that does not depend on ckeditor5-angular
    "tslib": "^1.9.0",
    "zone.js": "~0.9.1"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.800.0",
    "@angular/cli": "~8.0.0",
    "@angular/compiler-cli": "~8.0.0",
    ...
    "ts-node": "~7.0.0",
    "typescript": "~3.4.3"
  }
}
```

#### tsconfig.json

```js
{
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": "./",
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "declaration": false,
    "module": "esnext",
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "importHelpers": true,
    "target": "es5", // I had es2015 previously - same error
    "allowSyntheticDefaultImports": true,
    "typeRoots": [
      "node_modules/@types"
    ],
    "lib": [
      "es2018",
      "dom"
    ]
  }
}

```

#### angular.json

```js
{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "v1": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:component": {
          "style": "scss"
        }
      },
      "root": "",
      "sourceRoot": "src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/v1",
            "index": "src/index.html",
            "main": "src/main.ts",
            "polyfills": "src/polyfills.ts",
            "tsConfig": "tsconfig.app.json",
            "preserveSymlinks": true,
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "src/assets/styles/globals.scss"
            ],
            "stylePreprocessorOptions": {
              "includePaths": [
                "src/assets/styles"
              ]
            },
            "scripts": [
              "node_modules/phaser-ce/build/custom/pixi.js",
              "node_modules/phaser-ce/build/custom/p2.js",
              "node_modules/phaser-ce/build/custom/phaser-split.js"
            ]
          },
          "configurations": {
            "production": {
              "fileReplacements": [
                {
                  "replace": "src/environments/environment.ts",
                  "with": "src/environments/environment.prod.ts"
                }
              ],
              "optimization": true,
              "outputHashing": "all",
              "sourceMap": {
                "scripts": true,
                "styles": true,
                "hidden": true
              },
              "extractCss": true,
              "namedChunks": false,
              "aot": true,
              "extractLicenses": true,
              "vendorChunk": false,
              "buildOptimizer": true,
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "5mb",
                  "maximumError": "10mb"
                }
              ]
            }
          }
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "options": {
            "browserTarget": "v1:build"
          },
          "configurations": {
            "production": {
              "browserTarget": "v1:build:production"
            },
            "dev-server": {
              "browserTarget": "v1:build:dev-server"
            }
          }
        },

        ...
      }
    }}
}

**Comment by ma2ciek** [comment]
Thanks! I'll try to investigate it soon.

My guess is the `allowSyntheticDefaultImports` config.

**Comment by Kashspike** [comment]
Hi @ma2ciek, I could reproduce this issue.. can you try changing the "sourceMap" in production configuration to this:
`"sourceMap": {
        "hidden": true,
         "scripts": true
}`
and run 'ng build --prod'

![Screen Shot 2020-09-11 at 6 35 11 PM](https://user-images.githubusercontent.com/19363509/92926286-978da600-f45d-11ea-84f8-6920e3a7f38a.png)
![Screen Shot 2020-09-11 at 6 34 51 PM](https://user-images.githubusercontent.com/19363509/92926291-9a889680-f45d-11ea-9fb8-d1c2775d79af.png)

**Comment by ma2ciek** [comment]
Thanks, @Kashspike!

I was able to reproduce the issue and it seems that updating all dependencies together with the `ng-packgr` library should fix that issue. Though, the fix will work only for Angular>=9

**Comment by ma2ciek** [comment]
The recently released https://www.npmjs.com/package/@ckeditor/ckeditor5-angular/v/2.0.1 should fix this issue for Angular 9.1+.

**Comment by bevndas** [comment]
Was having the same issue for Angular 8, was using ckeditor-angular 2.1.0, ckeditor-27.1.0, - solved it by downgrading ckeditor-build-classic to @27.0.0
